### PR TITLE
ucx_info: fixes for RH1667548

### LINKED ISSUE #3147: fix ucx_info help message
https://bugzilla.redhat.com/show_bug.cgi?id=1667548

### PR DESCRIPTION
**Pull request by amaslenn**
Fixes issues described in https://bugzilla.redhat.com/show_bug.cgi?id=1667548.
Fixes #3147.

### CODE REVIEW

File: src/tools/info/ucx_info.c [hunk 1]
```diff
@@ -100,6 +100,7 @@ int main(int argc, char **argv)
             print_opts |= PRINT_UCP_EP;
             break;
         case 't':
+            print_opts |= PRINT_DEVICES;
```
Comment (yosefe) [inline src/tools/info/ucx_info.c]: maybe better update help message to indicate '-d' is needed for '-t' ?

File: src/tools/info/ucx_info.c [hunk 2]
```diff
@@ -183,7 +184,7 @@ int main(int argc, char **argv)
     }
 
     if (print_opts & (PRINT_UCP_CONTEXT|PRINT_UCP_WORKER|PRINT_UCP_EP)) {
-        if (ucp_features == 0) {
+        if (ucp_features == 0 && ucp_ep_params.field_mask == 0) {
```
Comment (yosefe) [inline src/tools/info/ucx_info.c]: 1. code style requires ( ) around each boolean sub-expression in && expression
2. need to update help and error message to indicate which flags in -u must be specified (e.g one of 'a', 'r', 't')

File: src/tools/info/ucx_info.c (hunk 1 above)
Comment (amaslenn) [reply src/tools/info/ucx_info.c]: Thought about it, but for the user this way is preferred. WDYT?

**Comment by mellanox-github** [comment]
Test PASSed.
See http://hpc-master.lab.mtl.com:8080/job/hpc-ucx-pr/8578/ for details (Mellanox internal link).

**Comment by mellanox-github** [comment]
Test PASSed.
See http://hpc-master.lab.mtl.com:8080/job/hpc-ucx-pr/8582/ for details (Mellanox internal link).

**Comment by yosefe** [comment]
@amaslenn can you port to v1.5.x as well?

**Pull request merged**